### PR TITLE
adding initial documentation on github actions

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: docs/Github/actions/githubactions.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+- https://github.com/actions/setup-java
+- https://github.com/eskatos/gradle-command-action
+
+# Workflows
+
+Github actions are workflows or if you prefer a set of jobs each with an associated list of tasks.  These jobs can be configured to run on specific platforms but typically run on linux.
+
+## Context
+
+Each gihtub workflow has a context via which you can access / set environment variables for use throughout your build.
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:30]: minor glitch: gihtub -> github

File: docs/Github/actions/githubactions.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# Github Actions
+
+## Introduction
+
+Android Identity uses Travis, Github Actions and Azure Devops to perform builds.  This document describes how we use github actions and provides examples and references for how github actions can be used to perform common tasks.  
+
+## References
+
+- Contexts & Expression Syntax: [https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:9]:
```suggestion
- [Contexts & Expression Syntax](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions)
```

File: docs/Github/actions/githubactions.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# Github Actions
+
+## Introduction
+
+Android Identity uses Travis, Github Actions and Azure Devops to perform builds.  This document describes how we use github actions and provides examples and references for how github actions can be used to perform common tasks.  
+
+## References
+
+- Contexts & Expression Syntax: [https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions)
+- Authentication in Github Actions: [https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:10]:
```suggestion
- [Authentication in GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow)
```

File: docs/Github/actions/githubactions.md (hunk 1 above)
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:30]:
```suggestion
Each GitHub workflow has a context via which you can access / set environment variables for use throughout your build.  

```

File: docs/Github/actions/githubactions.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+## Introduction
+
+Android Identity uses Travis, Github Actions and Azure Devops to perform builds.  This document describes how we use github actions and provides examples and references for how github actions can be used to perform common tasks.  
+
+## References
+
+- Contexts & Expression Syntax: [https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions)
+- Authentication in Github Actions: [https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow)
+- Github App Token Permissions: [https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:11]:
```suggestion
- [GitHub App Token Permissions](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token)
```

File: docs/Github/actions/githubactions.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+- Github Encrypted Secrets: [https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:12]:
```suggestion
- [GitHub Encrypted Secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets)
```

File: docs/Github/actions/githubactions.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+- Creating a personal access token: [https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:13]:
```suggestion
- [Creating a Personal Access Token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token)
```

File: docs/Github/actions/githubactions.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+- Azure DevOps Personal Access Token: [https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:14]:
```suggestion
- [Azure DevOps Personal Access Token](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page)
```

File: docs/Github/actions/githubactions.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+The following are used in github actions in use with Android Identity projects:
+
+- https://github.com/actions/checkout
+- https://github.com/actions/setup-java
+- https://github.com/eskatos/gradle-command-action
+
+# Workflows
+
+Github actions are workflows or if you prefer a set of jobs each with an associated list of tasks.  These jobs can be configured to run on specific platforms but typically run on linux.
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:26]:
```suggestion
GitHub actions are workflows or if you prefer a set of jobs each with an associated list of tasks.  These jobs can be configured to run on specific platforms but typically run on Linux.
```

File: docs/Github/actions/githubactions.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+## Secrets
+
+We currently have the following secrets configured within the common repo:
+
+[https://github.com/AzureAD/microsoft-authentication-library-common-for-android/settings/secrets/actions](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/settings/secrets/actions)
+
+### ACTION_PAT
+
+This is a github personal access token (PAT) based on the shoatman@microsoft.com account.  The PAT is configured with the following github OAuth2 permissions: public_repo, read:user, repo:status, user:email
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:98]:
```suggestion
This is a GitHub Personal Access Token (PAT) based on the [shoatman@microsoft.com](mailto:shoatman@microsoft.com) account.  The PAT is configured with the following GitHub OAuth2 permissions: 
- `public_repo`
- `read:user`
- `repo:status`
- `user:email`
```

File: docs/Github/actions/githubactions.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+- Creating a personal access token: [https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token)
+- Azure DevOps Personal Access Token: [https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page)
+
+### Actions
+
+The following are used in github actions in use with Android Identity projects:
+
+- https://github.com/actions/checkout
+- https://github.com/actions/setup-java
+- https://github.com/eskatos/gradle-command-action
```
Comment (AdamBJohnsonx) [inline docs/Github/actions/githubactions.md:22]:
```suggestion
- https://github.com/eskatos/gradle-command-action
- https://github.com/dangoslen/changelog-enforcer@v1.4.0

```

File: docs/Github/actions/githubactions.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+### ACTION_PAT
+
+This is a github personal access token (PAT) based on the shoatman@microsoft.com account.  The PAT is configured with the following github OAuth2 permissions: public_repo, read:user, repo:status, user:email
+
+> NOTE: We need to enable SSO by clicking the enable SSO button.  I noticed that on first use of the token within a Github action it was necessary to do this again using an error link in the output from the first run.  After that the token worked as expected.  So I'm assuming that the button didn't really do what I was hoping it would.
+
+### IDDP_PIPELINE
+
+This is an Azure DevOps personal access token based on the shoatman@microsoft.com account.  The PAT is configured with the folloiwng github OAuth2 permissions:
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:104]:
```suggestion
This is an Azure DevOps Personal Access Token based on the [shoatman@microsoft.com](mailto:shoatman@microsoft.com) account.  The PAT is configured with the following GitHub OAuth2 permissions: 

```

File: docs/Github/actions/githubactions.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+
+This is a github personal access token (PAT) based on the shoatman@microsoft.com account.  The PAT is configured with the following github OAuth2 permissions: public_repo, read:user, repo:status, user:email
+
+> NOTE: We need to enable SSO by clicking the enable SSO button.  I noticed that on first use of the token within a Github action it was necessary to do this again using an error link in the output from the first run.  After that the token worked as expected.  So I'm assuming that the button didn't really do what I was hoping it would.
+
+### IDDP_PIPELINE
+
+This is an Azure DevOps personal access token based on the shoatman@microsoft.com account.  The PAT is configured with the folloiwng github OAuth2 permissions: 
+
+- Build Scopes: [https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes](https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes)
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:106]:
```suggestion
- [Build Scopes](https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes):
```

File: docs/Github/actions/githubactions.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+- vso.build, vso.build_execute
```
Comment (iambmelt) [inline docs/Github/actions/githubactions.md:107]:
```suggestion
- `vso.build`
- `vso.build_execute`
```

**Review by iambmelt — APPROVED** [review]
Left some suggestions

File: docs/Github/actions/githubactions.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+```yaml
+
+# Here's an example of getting the github pat
+${{ secrets.ACTION_PAT }}
+
+```
+
+## Jobs
+
+Jobs are a grouping of tasks.  Jobs by default run in parallel, but can be coordinated to run sequentially (or in an order of your preference) by establishing job depdencies via the "needs:" yaml entry.
```
Comment (AdamBJohnsonx) [inline docs/Github/actions/githubactions.md:59]: Multiple dependencies can be specified with json list ("[a, b]") syntax.

**Pull request merged**